modules/compute/gke-node-pool/variables.tf

```diff
@@ -79,16 +79,32 @@ variable "guest_accelerator" {
   type = list(object({
     type  = optional(string)
     count = optional(number, 0)
-    gpu_driver_installation_config = optional(list(object({
+    gpu_driver_installation_config = optional(object({
       gpu_driver_version = string
-    })))
+    }), { gpu_driver_version = "DEFAULT" })
     gpu_partition_size = optional(string)
-    gpu_sharing_config = optional(list(object({
-      gpu_sharing_strategy       = optional(string)
-      max_shared_clients_per_gpu = optional(number)
-    })))
+    gpu_sharing_config = optional(object({
+      gpu_sharing_strategy       = string
+      max_shared_clients_per_gpu = number
+    }))
   }))
-  default = null
+  default  = []
+  nullable = false
+
+  validation {
+    condition     = alltrue([for ga in var.guest_accelerator : ga.count != null])
+    error_message = "var.guest_accelerator.count cannot be null"
+  }
+
+  validation {
+    condition     = alltrue([for ga in var.guest_accelerator : ga.count >= 0])
+    error_message = "var.guest_accelerator.count must never be negative"
+  }
+
+  validation {
+    condition     = alltrue([for ga in var.guest_accelerator : ga.gpu_driver_installation_config != null])
+    error_message = "var.guest_accelerator.gpu_driver_installation_config must not be set to null; use default value of empty list or set explicitly"
```

Comment:
I see a problem with this error message that we should resolve before merging. I have 3 tests running on the current code and I'd like to see their success/failure first.

Reply:
are these new validations ? Also maybe you want to update this as the test ran ?

Reply:
Yes, they are new, but they are valuable. In order of value:

## Protection against count = null

I have made the choice to refactor as little as possible in main.tf where it is not necessary. I think this code was written without a clear understanding of the behavior of `try(null, any, other, set, of values)`. The preceding code will return `null`. For example, this code (present in the PR and before the PR) will fail if count is null:

```hcl
  count = coalesce(try(ga.value.count, 0) > 0 ? ...
```

Proof:

```
~❯❯❯ terraform console
> try(null, 0) > 0
╷
│ Error: Operation failed
│
│   on <console-input> line 1:
│   (source code not available)
│
│ Error during operation: argument must not be null.
```

## Protection against count being negative

This is just sanity checking and I am doing it "while we are at it"

## Protection against gpu_driver_installation_config being null

The current code makes the choice to always install some driver, implicitly inserting a request for "DEFAULT" to GKE. I am choosing to make that default value more obvious in the variable definition, but otherwise the intent of this PR is to be a direct refactor for TPG 6.x without any behavioral changes.

IMO, we should allow the user to set this to null (so that no driver is installed) but that should be a small, focused PR.